lib/elixir/lib/calendar.ex

```diff
@@ -151,10 +228,11 @@ defmodule Date do
       true
 
   """
-  @spec utc_today() :: t
-  def utc_today() do
+  @spec utc_today(Calendar.calendar) :: t
+  def utc_today(calendar \\ Calendar.ISO) do
     {:ok, {year, month, day}, _, _} = Calendar.ISO.from_unix(:os.system_time, :native)
-    %Date{year: year, month: month, day: day}
+    with {:ok, date} <- new(year, month, day, Calendar.ISO),
+         do: convert!(date, calendar)
```

Comment:
This will fail if the calendar has a different rollover date (but it shouldn't because it is UTC).

If you check the from_unix function it is doing a conversion to rata die. So we probably want to do this:

  1. unix to rata die
  2. rata die to calendar
  3. get only the date fields

You will need to do similar in the `utc_*` functions in the other modules.